lib/ex_aws/utils.ex

```diff
@@ -109,23 +116,58 @@ defmodule ExAws.Utils do
     end)
   end
 
-  def build_indexed_params(key_template, values) when is_list(values) do
-    case String.split(key_template, "{i}") do
-      [prefix, suffix] -> 
-        values
-        |> Enum.with_index(1)
-        |> Enum.map(fn {value, i} -> {prefix <> "#{i}" <> suffix, value} end)
 
-      _ -> raise ArgumentError, "The Argument key_template is invalid. 
-      Expected a string with exactly one location for an index, got: \"#{key_template}\"
-      Example valid key_template: \"Tags.member.{i}.Key\""
-    end
+  # NOTE: flatten_params is not tail call optimized 
+  # but it is unlikely that any AWS params will ever
+  # be nested enough for this to  cause a stack overflow
+  def flatten_params(key, values) when is_list(values) do
+    values
+    |> Stream.with_index(1)
+    |> Stream.map(fn {value, i} -> {"#{maybe_camelize(key)}.#{i}", value} end)
+    |> Stream.flat_map(fn
+      {key, kv_pairs} when is_list(kv_pairs) -> 
+        add_prefix(key, kv_pairs) |> flatten_params
+        
+      {key, value} -> 
+        [{key, value}]
+    end)
+    |> Enum.to_list
+  end
+  # When only one key_template and value is passed  
+  def flatten_params(key, value), 
+  do: [{maybe_camelize(key), value}]
```

Comment:
Sorry, but I'm really not a fan of this style. Please either make it on a single line or, if it's going to be multiline, use `do end`